cpp/recognition/cograph/BretscherCorneilHabibPaulCographRecognition.cpp

```diff
@@ -0,0 +1,212 @@
+                previous_list[v] = in_which_list[v];
+                if (used[v]) {
+                    continue;
+                }
+                auto number = in_which_list[v];
+                if (slice_size != 1 && number == in_which_list[x]) {
+                    SA++;
+                }
+                auto previous = number;
+
+                auto copy_number = number;
+                auto end_it = L.end();
+                end_it--;
+
+                if (is_complement) {
+                    copy_number++;
+                }
+
+
+                if ((number == L.begin() && !is_complement) ||
+                (number == end_it && is_complement)) {
+                    std::list<NetworKit::node> insert;
+                    L.insert(copy_number, insert);
+                }
+                if (!is_complement) {
+                    previous--;
+                } else {
+                    previous++;
+                }
+                if (!previous->empty() && (!graph.hasEdge(previous->front(), x)
+                                           || !used_at_this_step[previous->front()] ||
+                                           previous_list[previous->front()] != previous_list[v])) {
+                    std::list<NetworKit::node> insert;
+                    L.insert(copy_number, insert);
+                    previous = number;
+                    if (is_complement) {
+                        previous++;
+                    } else {
+                        previous--;
+                    }
+                }
+                previous->push_back(v);
+                in_which_list[v] = previous;
+                number->erase(in_which_position[v]);
+                if (number->empty()) {
+                    L.erase(number);
+                }
+
+                auto end = previous->end();
+                end--;
+                in_which_position[v] = end;
+            }
+            if (is_complement) {
+                SA = slice_size - 1 - SA;
+            }
+            for (auto v : graph.neighborRange(x)) {
+                used_at_this_step[v] = false;
+            }
+            if (slice_size - SA - 1 != 0) {
+                what_ends_here[i + SA].emplace_back(x, slice_size - SA - 1);
+            }
+        }
+        return {borders, ans};
+    }
+
+    bool BretscherCorneilHabibPaulCographRecognition::NeighbourhoodSubsetProperty
+    (bool is_complement, std::vector<NetworKit::node> a,
+    std::vector<std::vector<std::pair<int, int>>> borders) {
+        std::vector<bool> used(a.size());
+        std::vector<int> pos(a.size());
+        for (unsigned int i = 0; i < a.size(); i++) {
+            pos[a[i]] = i;
+        }
+        for (auto i : a) {
+            if (borders[i].empty()) {
+                continue;
+            }
+            for (unsigned int j = 0; j < borders[i].size() - 1; j++) {
+                auto l = borders[i][j].first;
+                auto r = borders[i][j].second;
```

Comment:
`auto [l, r] = borders[i][j];` should work

Reply:
fixed